curriculum/challenges/english/25-front-end-development/lecture-working-with-higher-order-functions-and-callbacks/673362e43d57b51f1ad2d466.md

```diff
@@ -10,6 +10,48 @@ dashedName: how-do-the-every-and-some-methods-work
 
 Watch the lecture video and answer the questions below.
 
+# --transcript
+
+How do the `every()` and `some()` methods work in JavaScript?
+
+When you're working with arrays in JavaScript, there are often times when you want to check if all elements in an array meet a certain condition, or if at least one element meets a condition.
+
+This is where the `every()` and `some()` methods come in handy. These methods are powerful tools that can simplify your code and make it more readable.
+
+Let's start with the `every()` method. This method tests whether all elements in an array pass a test implemented by a provided function. In simpler terms, it checks if every single item in your array satisfies a condition you specify.
+
+The `every()` method returns true if the provided function returns true for all elements in the array. If any element fails the test, the method immediately returns false and stops checking the remaining elements.
+
+Here's an example to illustrate how `every()` works:
+
+```js
+const numbers = [2, 4, 6, 8, 10];
+const hasAllEvenNumbers = numbers.every((num) => num % 2 === 0);
+
+console.log(hasAllEvenNumbers); // true
+```
+
+In this example, we're checking if all numbers in the array are even. The function we provide to `every()` checks if each number is divisible by `2` with no remainder. Since all numbers in our array are indeed even, `hasAllEvenNumbers` will be true.
+
+Now, let's look at the `some()` method.
+
+While `every()` checks if all elements pass a test, `some()` checks if at least one element passes the test. The `some()` method returns true as soon as it finds an element that passes the test. If no elements pass the test, it returns false.
+
+Here's an example of how `some()` works:
+
+```js
+const numbers = [1, 3, 5, 7, 8, 9];
+const hasSomeEvenNumbers = numbers.some((num) => num % 2 === 0);
+
+console.log(hasSomeEvenNumbers); // true
+```
+
+In this example, we're checking whether any number in the array is even. The function we pass to `some()` is the same as before. Even though most numbers in our array are odd, `hasEven` will be true because there's at least one even number (`8`) in the array.
```

Comment:
```suggestion
In this example, we're checking whether any number in the array is even. The function we pass to `some()` is the same as before. Even though most numbers in our array are odd, `hasEven` will be `true` because there's at least one even number (`8`) in the array.
```